webui/react/src/stores/workspaces.tsx

```diff
@@ -53,15 +60,40 @@ export const useEnsureWorkspacesFetched = (canceler: AbortController): (() => Pr
     } catch (e) {
       handleError(e);
     }
-  }, [canceler, workspaces, updateWorkspaces]);
+  }, [canceler, workspaces]);
 };
 
-export const useWorkspaces = (): Loadable<Workspace[]> => {
+export const useWorkspaces = (params?: GetWorkspacesParams): Loadable<Workspace[]> => {
   const context = useContext(WorkspacesContext);
   if (context === null) {
     throw new Error('Attempted to use useWorkspaces outside of Workspace Context');
   }
-  const { workspaces } = context;
+  if (context.workspaces === NotLoaded) {
+    return NotLoaded;
+  }
+  const val = Loadable.getOrElse([], context.workspaces).filter((ws) =>
+    Object.keys(params || {}).reduce<boolean>((accumulator: boolean, key) => {
+      switch (key) {
+        case 'archived':
+          return accumulator && ws.archived === params?.archived;
+        case 'pinned':
+          return accumulator && ws.pinned === params?.pinned;
+        case 'name':
+          return accumulator && ws.name === params?.name;
+        case 'users':
+          return accumulator && (params?.users || []).indexOf(String(ws.userId)) > -1;
+        default:
+          return false;
+      }
+    }, true),
+  );
+  return Loaded(val);
+};
 
-  return workspaces;
+export const updateWorkspace = (id: number, updater: (arg0: Workspace) => Workspace): void => {
```

Comment:
This isn't very useful as is, you'll want to make this like the `useFetchWorkspaces` and return a function that actually does the updating, that way you can call this from a callback.